lib/api/core/httpRouter/index.js

```diff
@@ -158,48 +158,56 @@ class Router {
           .catch(error => replyWithError(cb, request, error));
       }
 
-      return replyWithError(cb, request, new BadRequestError(`Unrecognized HTTP method ${httpRequest.method}`));
+      return Promise.reject(replyWithError(cb, request, new BadRequestError(`Unrecognized HTTP method ${httpRequest.method}`)));
     }
 
     httpRequest.url = httpRequest.url.replace(LeadingSlashRegex, '');
 
     try {
       const routeHandler = this.routes[httpRequest.method].getHandler(httpRequest);
+      const httpEvent = `http:${httpRequest.method.toLowerCase()}`;
 
       request = routeHandler.getRequest();
       request.response.setHeaders(this.defaultHeaders);
 
-      if (routeHandler.handler === null) {
-        return replyWithError(cb, request, new NotFoundError(`API URL not found: ${routeHandler.url}`));
-      }
+      return this.kuzzle.pluginsManager.trigger(httpEvent, request)
+        .then(mutatedRequest => {
+          routeHandler.request = mutatedRequest;
 
-      if (httpRequest.content.length <= 0) {
-        return routeHandler.invokeHandler(cb);
-      }
+          if (routeHandler.handler === null) {
```

Comment:
should be moved before the event trigger, to avoid calling plugin pipes if the URL does not match any route.

Reply:
Are you sure ? Even if the request does not match any route, it's still a HTTP request received by Kuzzle, shouldn't we trigger the event anyway ?